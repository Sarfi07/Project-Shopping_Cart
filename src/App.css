#root {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

header,
footer {
  flex-shrink: 0;
}

main {
  flex: 1;
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
  justify-content: center;
  padding-top: 4rem;
}

.totalAmount {
  width: 100%;
  text-align: center;
  font-weight: bold;
  font-size: 2rem;
}

.checkoutLink,
.payBtn {
  text-align: center;
  text-decoration: none;
  color: white;
  font-size: 1.5rem;
  border-radius: 0.5rem;
  width: 30%;
  background-color: rgb(180, 111, 26);
}

.payBtn {
  height: fit-content;
}

.homePage {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 70vh;
  padding: 2rem;

  gap: 2rem;
  text-align: center;

  img {
    flex: 1;
    max-width: 50%;
    height: 70%;
  }

  p {
    flex: 1;
    padding: 20px;
    font-size: 1.2rem;
    text-align: left;
  }
}

@media only screen and (max-width: 768px) {
  .homePage {
    flex-direction: column;

    img {
      max-width: 80%;
      margin: 0 auto;
    }

    p {
      text-align: center;
    }
  }
}
