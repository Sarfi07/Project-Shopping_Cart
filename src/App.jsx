import './App.css';
import Header from './components/header/Header';
import { Outlet } from 'react-router-dom';
import { useState } from 'react';

function App() {
  const [cart, setCart] = useState([]);

  function addToCart(product, quantity) {
    const newObj = {
      item: product,
      quantity: parseInt(quantity),
    };
    const updatedCart = [...cart, newObj];
    setCart(updatedCart);
    console.log(updatedCart);
  }

  function removeFromCart(product) {
    const productIndex = cart.findIndex((item) => item.item.id === product.id);

    if (productIndex >= 0) {
      const updatedCart = [...cart];
      updatedCart.splice(productIndex, 1);

      setCart(updatedCart);
    } else {
      console.log('Product not found in array');
    }
  }

  const contextValue = { cart, addToCart, removeFromCart, setCart };
  return (
    <>
      <Header cartLength={cart.length} />
      <main>
        <Outlet context={contextValue} />
      </main>
    </>
  );
}

export default App;
