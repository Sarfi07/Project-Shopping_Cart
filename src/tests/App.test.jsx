import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';

import App from '../App';
import { BrowserRouter } from 'react-router-dom';
// import { createContext, useReducer } from 'react';
// import userEvent from '@testing-library/user-event';
// import { MemoryRouter, Routes, Route } from 'react-router-dom';
// import HomePage from '../components/HomePage';
// import ShoppingPage from '../components/ShoppingCart';
// import Cart from '../components/Cart';

// const CartContext = createContext();
// // const renderWithContext = (contextValue, initialPath = '/') => {
// //   return render(
// //     <MemoryRouter initialEntries={[initialPath]}>
// //       <CartContext.Provider value={contextValue}>
// //         <Routes>
// //           <Route path="/" element={<App />}>
// //             <Route index element={<HomePage />} />
// //             <Route path="/shop" element={<ShoppingPage />} />
// //             <Route path="/cart" element={<Cart />} />
// //           </Route>
// //         </Routes>
// //       </CartContext.Provider>
// //     </MemoryRouter>,
// //   );
// // };

const renderWithRouter = (component) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

const cart = [];
const addToCart = vi.fn((product, quantity) => {
  fakeContext.cart.push({ product, quantity });
});
const removeFromCart = vi.fn();
const setCart = vi.fn();

// const proudct = {
//   product: {
//     id: 1,
//     item: 'Laptop',
//     price: 110,
//   },
//   quantity: 1,
// };

const fakeContext = { cart, addToCart, removeFromCart, setCart };

describe('App Components Tests', () => {
  it('should components', async () => {
    const { container } = renderWithRouter(<App context={fakeContext} />);

    expect(container).toMatchSnapshot();
  });

  // it('Add to Cart', async () => {
  //   const user = userEvent.setup();

  //   renderWithContext(fakeContext, '/shop');

  //   const addToCartBtn = screen.getByText(/Add to Cart/i);
  //   await user.click(addToCartBtn);

  //   expect(fakeContext.cart.length).toBe(1);
  // });
});
