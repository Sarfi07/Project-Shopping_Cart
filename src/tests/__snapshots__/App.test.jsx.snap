// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Components Tests > should add items to the cart 1`] = `
<div>
  <header>
    <h1>
      Shopping Cart
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          Cart (
          0
          )
        </a>
      </li>
    </ul>
  </header>
  <main />
</div>
`;

exports[`App Components Tests > should components 1`] = `
<div>
  <header>
    <h1>
      Shopping Cart
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          Cart (
          0
          )
        </a>
      </li>
    </ul>
  </header>
  <main />
</div>
`;
